pkg/test/masteruserrecord/assertion.go

```diff
@@ -92,6 +92,42 @@ func (a *Assertion) AllUserAccountsHaveCondition(expected toolchainv1alpha1.Cond
 	return a
 }
 
+func (a *Assertion) AllUserAccountsHaveTier(tierName string, namespaces []toolchainv1alpha1.NSTemplateTierNamespace) *Assertion {
```

Comment:
Why not just pass NSTemplateTier here?
Something like:
```suggestion
func (a *Assertion) AllUserAccountsHaveTier(tier NSTemplateTier) *Assertion {
```

Reply:
yes, I was thinking about providing the whole `NSTemplateTier` object, but I decided for two params just for the sake of the simplicity of calling this assertion method. The reason is that setting the name in the `NSTemplateTier` is not obvious and requires creation of `ObjectMeta` object that requires additional lines of code.

Reply:
ok, looking at all usages of the assertion methods I have to admit that I can easily pass the whole `NSTemplateTier` object. So changed 262d7bb